internal/handshake/xor_nonce_aead_boring.go

```diff
@@ -0,0 +1,51 @@
+//go:build boringcrypto
+
+package handshake
+
+import (
+	"crypto/cipher"
+	"crypto/tls"
+	"os"
+	"strings"
+)
+
+var goBoringDisabled bool = strings.TrimSpace(os.Getenv("QUIC_GO_DISABLE_BORING")) == "1"
```

Comment:
Do we want to keep this?

Reply:
Let's keep it for now. We can remove this after we verify the rebased version works.